x-pack/plugins/security/server/authorization/privileges/privileges.ts

```diff
@@ -47,15 +47,15 @@ export function privilegesFactory(
           predicate: (pId, featurePrivilege) => !featurePrivilege.excludeFromBasePrivileges,
         })) {
           const privilegeActions = featurePrivilegeBuilder.getActions(privilege, feature);
-          allActions = [...allActions, ...privilegeActions];
+          privilegeActions.forEach((action) => allActionsSet.add(action));
           if (privilegeId === 'read') {
-            readActions = [...readActions, ...privilegeActions];
+            privilegeActions.forEach((action) => readActionsSet.add(action));
           }
```

Comment:
__nit:__ since you have `forEach` already, maybe we can combine these:
```js
privilegeActions.forEach((action) => {
  allActionsSet.add(action);
  if (privilegeId === 'read') {
    readActionsSet.add(action);
  }
});
```